packages/apollo/src/commands/service/check.ts

```diff
@@ -253,112 +260,198 @@ export default class ServiceCheck extends ProjectCommand {
+                  configName
+                )}`;
+
+                if (errors.length > 0) {
+                  const decodedErrors = errors.filter(Boolean).map(error => {
+                    // We `filter` by `Boolean` above; this will never happen and is just here to appease
+                    // TypeScript.
+                    if (!error) {
+                      return {};
+                    }
+
+                    const match = error.message.match(
+                      /^\[([^\[]+)\]\s+(\S+)\ ->\ (.+)/
+                    );
+
+                    if (!match) {
+                      // If we can't match the errors, that means they're in a format we don't recognize.
+                      // Report the entire string as the user will see the raw message.
+                      return { message: error.message };
                     }
+
+                    const [, service, field, message] = match;
+                    return { service, field, message };
                   });
 
-                  // FIXME: reformat to match other check results
+                  let result = "";
+
+                  cli.table(decodedErrors, {
+                    columns: [
+                      { key: "service", label: "Service" },
+                      { key: "field", label: "Field" },
+                      { key: "message", label: "Message" }
+                    ],
+                    // The default `printLine` will output to the console; we want to capture the output so we
+                    // can test it.
+                    printLine: line => {
+                      result += `\n${line}`;
+                    }
+                  });
 
-                  taskOutput.federation = {
-                    errors,
-                    warnings
-                  };
+                  this.log(result);
 
-                  if (compositionValidationDetails) {
-                    schemaHash = compositionValidationDetails.schemaHash;
-                  } else {
-                    // FIXME: We should provide some better error handling at the GraphQL layer for this
-                    throw new Error(`Federated service could not be composed due to the following errors:
-  ${errors && errors.map(err => (err && err.message) || "").join("\n")}
-                  `);
-                  }
-                } else {
-                  schema = await project.resolveSchema({ tag: config.tag });
+                  // We have to throw an error here even though we already reported the errors in a table
+                  // because we need to short-circuit the rest of the tasks in the service check.
+                  throw new Error(
+                    federatedServiceCompositionUnsuccessfulErrorMessage
+                  );
+                }
+              }
+            },
+            {
+              title: `Validating ${
+                serviceName ? "composed" : "local"
```

Comment:
I realized that local might not be accurate depending on the source/endpoint. Would it be reasonable to remove it?

Reply:
That makes sense to me. Just to clarify:

For federated services we'll say: "Validating composed schema..."

For non-federated services we'll say: "Validating schema..."

Does that sound right?

Reply:
Yeah! That sounds right